File: resources/expected/ros2/fan_in_fan_out/lax/src/fan_in_fan_out_system_cpp_pkg/src/base_code/fanIn_consumer_base_src.cpp
#include "fan_in_fan_out_system_cpp_pkg/base_headers/fanIn_consumer_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

fanIn_consumer_base::fanIn_consumer_base() : Node("fanIn_consumer")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    fanIn_consumer_myInteger_subscription_ = this->create_subscription<fan_in_fan_out_system_cpp_pkg_interfaces::msg::Integer64>(
        "fanIn_consumer_myInteger",
        1,
        std::bind(&fanIn_consumer_base::handle_myInteger, this, std::placeholders::_1), subscription_options_);

}

File: resources/expected/ros2/python-fan_in_fan_out/lax/src/fan_in_fan_out_system_py_pkg/fan_in_fan_out_system_py_pkg/base_code/fanIn_consumer_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from fan_in_fan_out_system_py_pkg.user_code.fanIn_consumer_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from fan_in_fan_out_system_py_pkg_interfaces.msg import Integer64

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class fanIn_consumer_base(Node):
    def __init__(self):
        super().__init__("fanIn_consumer")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.fanIn_consumer_myInteger_subscription_ = self.create_subscription(
            Integer64,
            "fanIn_consumer_myInteger",
            self.handle_myInteger,
            1,
            callback_group=self.cb_group_)

    #=================================================
    #  C o m p u t e    E n t r y    P o i n t
    #=================================================
    def handle_myInteger(self, msg):
        raise NotImplementedError("Subclasses must implement this method")

